xtask/src/ci/spec_links.rs

```diff
@@ -43,61 +44,34 @@ fn walk_dirs(path: &Path, split: &str, version_match: fn(&str) -> bool) -> Resul
                 let content = fs::read_to_string(&path)?;
                 let split = content.split(split).collect::<Vec<_>>();
 
-                match split.as_slice() {
-                    // No spec link was found
-                    // `[]` is only to satisfy exhaustiveness checking
-                    [] | [_] => {}
-                    [pre, post, rest @ ..] => {
-                        if pre.lines().rev().next().map_or(false, |l| l.starts_with("//!"))
-                            && !version_match(post)
-                        {
-                            return Err(format!(
-                                "error: spec URL with no version number\nfrom: {}\n{}",
-                                path.display(),
-                                &post[0..25]
-                            )
-                            .into());
+                let mut last_segment = false;
```

Comment:
I don't understand this variable, it seems more like "not first" rather than last considering the first loop iteration sets it to `true`. Maybe we can do away with it by just assuming that there's never going to be two spec links on the same line, and working off of the file's lines (can use `BufReader::new(File::new(&path)?)` then instead of loading the whole file into memory)? I think that lead to easier code than the whole "split by the URL" thing.

Reply:
If I switch to BufReader` I would still be splitting by the URL just on each line right?

The variable `last_segment` is to stop erroring on files that contain no spec links, so calling it `no_spec_link` might make more sense?

Reply:
> If I switch to BufReader` I would still be splitting by the URL just on each line right?

I'd personally use find + indexing stuff.. I guess that's not really easier though, hm..

> The variable `last_segment` is to stop erroring on files that contain no spec links, so calling it `no_spec_link` might make more sense?

This is where I feel like `find` just makes a lot more sense, if you don't find the string then just do nothing (rather than having to separately track whether `split` returned the entire input).